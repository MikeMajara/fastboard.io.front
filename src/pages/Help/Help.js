import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';

import { HeadMeta } from '#components';
import routes from '#routes';
import { analyticsService } from '#services';

import createImg from '#theme/images/help-create.png';
import joinImg from '#theme/images/help-join.png';
import settingsImg from '#theme/images/help-settings.png';
import leaveImg from '#theme/images/help-leave.png';

import drawingImg from '#theme/images/help-drawing.png';
import selectionImg from '#theme/images/help-selection.png';
import writingImg from '#theme/images/help-writing.png';
import pointerImg from '#theme/images/help-pointer.png';
import saveRemoveImg from '#theme/images/help-save-remove.png';

import {
  SWrapper,
  SLine,
  SHeader,
  SSubHeader,
  SContent,
  SSection,
  SSectionImage,
  SSectionText,
  SSectionTextHeader,
} from './styled';

const Help = () => {
  const { t } = useTranslation('help');

  // Analytics pageview
  useEffect(() => analyticsService.pageview(routes.help), []);

  return (
    <>
      <HeadMeta route={routes.help} />
      <SWrapper>
        <SHeader>
          {t('title')}
        </SHeader>

        <SLine />

        <SSubHeader>
          {t('sessions.title')}
        </SSubHeader>
        <SContent>
          {t('sessions.tagline')}
        </SContent>
        <SContent extraSpace>
          <SSection>
            <SSectionImage image={createImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('sessions.create.header')}
              </SSectionTextHeader>
              <div>
                {t('sessions.create.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={joinImg} invert />
            <SSectionText>
              <SSectionTextHeader>
                {t('sessions.join.header')}
              </SSectionTextHeader>
              <div>
                {t('sessions.join.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={settingsImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('sessions.settings.header')}
              </SSectionTextHeader>
              <div>
                {t('sessions.settings.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={leaveImg} invert />
            <SSectionText>
              <SSectionTextHeader>
                {t('sessions.leave.header')}
              </SSectionTextHeader>
              <div>
                {t('sessions.leave.content')}
              </div>
            </SSectionText>
          </SSection>
        </SContent>

        <SLine />

        <SSubHeader>
          {t('tools.title')}
        </SSubHeader>
        <SContent>
          {t('tools.tagline')}
        </SContent>
        <SContent extraSpace>
          <SSection>
            <SSectionImage image={drawingImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('tools.drawing.header')}
              </SSectionTextHeader>
              <div>
                {t('tools.drawing.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={writingImg} invert />
            <SSectionText>
              <SSectionTextHeader>
                {t('tools.writing.header')}
              </SSectionTextHeader>
              <div>
                {t('tools.writing.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={selectionImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('tools.manipulating.header')}
              </SSectionTextHeader>
              <div>
                {t('tools.manipulating.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={pointerImg} invert />
            <SSectionText>
              <SSectionTextHeader>
                {t('tools.pointer.header')}
              </SSectionTextHeader>
              <div>
                {t('tools.pointer.content')}
              </div>
            </SSectionText>
          </SSection>

          <SSection>
            <SSectionImage image={saveRemoveImg} />
            <SSectionText>
              <SSectionTextHeader>
                {t('tools.saveOrDelete.header')}
              </SSectionTextHeader>
              <div>
                {t('tools.saveOrDelete.content')}
              </div>
            </SSectionText>
          </SSection>
        </SContent>
      </SWrapper>
    </>
  );
};

export default Help;
