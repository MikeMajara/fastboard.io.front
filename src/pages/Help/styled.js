import styled, { css, breakpoints, up } from '@xstyled/styled-components';
import { getSpace } from '@xstyled/system';

// @todo refactor this - with About section
export const SWrapper = styled.div`
  margin: auto;
  text-align: justify;
  max-width: miscSectionWidth;

  ${breakpoints({
    xs : css`padding: default;`,
    sm : css`padding: 40px;`,
    md : css`padding: 40px 80px;`,
    lg : css`padding: 40px 120px;`,
  })}
`;

export const SLine = styled.hr`
  border-width: default;
  border-style: solid;
  border-color: border;
  margin-top: xl;
  margin-bottom: xl;
`;

export const SHeader = styled.h1`
  margin-top: default;
  margin-bottom: default;
  font-size: 40px;
  text-align: center;
`;

export const SSubHeader = styled.h2`
  margin-top: default;
  margin-bottom: default;
  font-size: xlg;
  text-align: center;
`;

export const SContent = styled.div`
  margin-top: ${({ extraSpace }) => (extraSpace ? getSpace('xl') : getSpace('default'))};
  text-align: center;
`;

export const SSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-top: default;

  ${up('md', css`
    height: 300px;
    flex-direction: row;
    margin-top: 60px;
  `)}
`;

export const SSectionImage = styled.div`
  height: 200px;
  background-image: url(${({ image }) => image});
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;

  ${up('md', css`
    width: 45%;
    height: 100%;
    order: ${({ invert }) => (invert ? '1' : 'unset')};
  `)}
`;

export const SSectionText = styled.div`
  box-sizing: border-box;
  padding: default;

  ${up('md', css`width: 45%;`)}
`;

export const SSectionTextHeader = styled.h3`
  font-size: xlg;
`;
