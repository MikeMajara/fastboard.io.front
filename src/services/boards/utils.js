/**
 * Send boards event trough the network.
 *
 * @param {String} eventName Event key name.
 * @param {Object} data Data associated to the event.
 * @param {Function} ask Callback for server event acknowledgement.
 */
function send(eventName, data, ack) {
  this.socket.emit(eventName, data, ack);
}

/**
 * Check if the service is initialized.
 *
 * @return {Boolean} True if the service is initialized; false otherwise.
 */
function isInit() {
  return this.isInit;
}

export {
  isInit,
  send,
};
