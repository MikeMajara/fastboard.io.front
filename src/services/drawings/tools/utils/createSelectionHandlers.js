import { Group, Shape } from 'paper';

import { Log } from '#utils';
import { cursorTypes, selectionColorCode } from '#constants';
import store, { setSelectorCursorHover } from '#store';

import { bounds } from '../selector/constants';

const boundsArray = Object.values(bounds);
const bounds2cursorType = Object.freeze({
  [bounds.topLeft]     : cursorTypes.rotating,
  [bounds.topRight]    : cursorTypes.menu,
  [bounds.bottomLeft]  : cursorTypes.resizingSW,
  [bounds.bottomRight] : cursorTypes.resizingSE,
});

/**
 * Add selection handlers to the selected item.
 *
 * @param {Object} selectedItem Item to be added selection handlers to.
 * @param {Object} selectionLayer Layer that will contain the selection handlers.
 *
 * @returns {Object} Item - group with 4 handler items - from the selection handler layer.
 */
export default function createSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : createSelectionHandlers', { selectedItem });

  // Set selected item to show its selection state
  selectedItem.set({
    selected      : true,
    selectedColor : selectionColorCode,
    onMouseEnter  : () => store.dispatch(setSelectorCursorHover(cursorTypes.dragging)),
    onMouseLeave  : () => store.dispatch(setSelectorCursorHover()),
  });

  // Create bound handlers on the selection layer
  boundsArray.forEach((bound) => {
    selectionLayer.addChild(new Shape.Circle({
      name         : bound,
      center       : selectedItem.internalBounds[bound],
      radius       : 6,
      fillColor    : selectionColorCode,
      onMouseEnter : () => store.dispatch(setSelectorCursorHover(bounds2cursorType[bound])),
      onMouseLeave : () => store.dispatch(setSelectorCursorHover()),
    }));
  });

  // Create a group with the handlers to manipulate all of them at once
  selectionLayer.addChild(
    new Group({
      name     : 'selectionHandlers', // @todo to constants
      children : boundsArray.map(
        (bound) => selectionLayer.children[bound],
      ),
    }),
  );

  // Reproduce the selected item transformations on the handler items group
  selectionLayer.children.selectionHandlers.transform(selectedItem.matrix);

  // Except the scaling of the handler items themselves, which retain the original size
  selectionLayer.children.selectionHandlers.children.forEach((handlerItem) => {
    handlerItem.set({ scaling : 1 });
  });

  return selectionLayer.children.selectionHandlers;
}
