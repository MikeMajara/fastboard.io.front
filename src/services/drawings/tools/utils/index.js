export { default as createSelectionHandlers } from './createSelectionHandlers';
export { default as removeSelectionHandlers } from './removeSelectionHandlers';
export { default as resizeSelectionHandlers } from './resizeSelectionHandlers';
export { default as rotateSelectionHandlers } from './rotateSelectionHandlers';
export { default as translateSelectionHandlers } from './translateSelectionHandlers';
