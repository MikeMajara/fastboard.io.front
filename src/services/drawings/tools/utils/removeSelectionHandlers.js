import { Log } from '#utils';

/**
 * Remove selection handlers.
 *
 * @param {Object} selectedItem Item to be removed selection handlers from.
 * @param {Object} selectionLayer Layer that contains the selection handlers.
 */
export default function removeSelectionHandlers(selectedItem, selectionLayer) {
  Log.debug('Service : Drawings : Tools : Utils : removeSelectionHandlers');

  selectedItem.set({ selected : false });
  selectionLayer.removeChildren();
}
