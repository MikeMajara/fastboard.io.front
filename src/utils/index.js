export { default as Log } from './logger';
export { default as generateId } from './generateId';
export { default as point2net } from './point2net';
export { default as setPreventUnload } from './setPreventUnload';
export { default as throttle } from './throttle';
export { default as timeoutPromise } from './timeoutPromise';
export { default as triggerDownload } from './triggerDownload';
export { default as triggerUpload } from './triggerUpload';
export { default as validBoardId } from './validBoardId';
