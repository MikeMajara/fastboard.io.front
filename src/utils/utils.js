/**
 * Throttles a function call.
 *
 * @param {Function} callback Function to be throttled.
 * @param {Integer} delay Throttle delay in milliseconds.
 *
 * @return {Function} Function that throttles the execution of the callback.
 */
export const throttle = (callback, delay = 0) => {
  let previousTime = new Date().getTime();

  return function throttledCallback(args) {
    const currentTime = new Date().getTime();

    if ((currentTime - previousTime) >= delay) {
      previousTime = currentTime;
      callback(...args);
    }
  };
};

/**
 * Creates a Promise with a rejection timeout.
 *
 * @param {Function} callback Function for the promise. Receives res and rej functions.
 * @param {Integer} timeout Timeout for the promise to be rejected in milliseconds.
 *
 * @return {Promise} Promise with rejection timeout.
 */
export const timeoutPromise = (callback, timeout) => new Promise(
  (res, rej) => {
    const timeoutId = setTimeout(() => rej(new Error()), timeout);

    callback(
      (...args) => {
        clearTimeout(timeoutId);
        res(...args);
      },
      (...args) => {
        clearTimeout(timeoutId);
        rej(new Error(...args));
      },
    );
  },
);
