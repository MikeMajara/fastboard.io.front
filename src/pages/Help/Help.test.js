import React from 'react';
import { HelmetProvider } from 'react-helmet-async';

import { render } from '#devTools/tests/utils';

import Help from './Help';

jest.mock('#services', () => ({
  analyticsService : {
    pageview : jest.fn(),
  },
}));

describe('Page : Help', () => {
  test('Snapshot', () => {
    const { asFragment } = render(
      <HelmetProvider>
        <Help />
      </HelmetProvider>,
      { router : true },
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
