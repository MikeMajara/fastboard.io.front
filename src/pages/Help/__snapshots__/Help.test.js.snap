// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page : Help Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-prQdK dobkOb"
  >
    <h1
      class="sc-qYhTA fgpbBw"
    >
      title
    </h1>
    <hr
      class="sc-qQlgh jSSILe"
    />
    <h2
      class="sc-pAYXY gVPaRf"
    >
      sessions.title
    </h2>
    <div
      class="sc-pJwpB bhJGlF"
    >
      sessions.tagline
    </div>
    <div
      class="sc-pJwpB dmrkMS"
    >
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gwcKFP"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            sessions.create.header
          </h3>
          <div>
            sessions.create.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gTalGT"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            sessions.join.header
          </h3>
          <div>
            sessions.join.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gwcKFP"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            sessions.settings.header
          </h3>
          <div>
            sessions.settings.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gTalGT"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            sessions.leave.header
          </h3>
          <div>
            sessions.leave.content
          </div>
        </div>
      </div>
    </div>
    <hr
      class="sc-qQlgh jSSILe"
    />
    <h2
      class="sc-pAYXY gVPaRf"
    >
      tools.title
    </h2>
    <div
      class="sc-pJwpB bhJGlF"
    >
      tools.tagline
    </div>
    <div
      class="sc-pJwpB dmrkMS"
    >
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gwcKFP"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            tools.drawing.header
          </h3>
          <div>
            tools.drawing.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gTalGT"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            tools.writing.header
          </h3>
          <div>
            tools.writing.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gwcKFP"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            tools.manipulating.header
          </h3>
          <div>
            tools.manipulating.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gTalGT"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            tools.pointer.header
          </h3>
          <div>
            tools.pointer.content
          </div>
        </div>
      </div>
      <div
        class="sc-pRSgm jeOlmm"
      >
        <div
          class="sc-qapxP gwcKFP"
        />
        <div
          class="sc-oTmHo cjgrgd"
        >
          <h3
            class="sc-pbJYR gOlCjp"
          >
            tools.saveOrDelete.header
          </h3>
          <div>
            tools.saveOrDelete.content
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
